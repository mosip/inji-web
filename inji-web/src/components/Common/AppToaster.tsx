import React from "react";
import {ToastContainer} from "react-toastify";

export const AppToaster: React.FC = () => {
    return <ToastContainer
        position="top-right"
        autoClose={1000}
        hideProgressBar
        newestOnTop
        closeOnClick
        rtl={false}
        icon={<React.Fragment/>}
        pauseOnFocusLoss
        draggable
        pauseOnHover
        style={{width: '400px'}}
        theme="colored"
    />
}
