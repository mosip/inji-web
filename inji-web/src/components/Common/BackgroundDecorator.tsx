import React from 'react';

interface BackgroundDecoratorProps {
    circleCount?: number;
    topOffset?: number;
    baseRadius?: number;
    logoSrc?: string;
    logoAlt?: string;
    testId?: string;
}

export const BackgroundDecorator: React.FC<BackgroundDecoratorProps> = ({
    circleCount = 6,
    topOffset = 155,
    baseRadius = 96,
    logoSrc,
    logoAlt = "Logo",
    testId = "background-decorator"
}) => {
    return (
        <div 
            className="overflow-hidden absolute inset-0 z-0 flex items-center justify-center pointer-events-none"
            data-testid={testId}
        >
            <div className={`absolute top-[${topOffset}px]`} style={{ top: `${topOffset}px` }}>
                {[...Array(circleCount)].map((_, index) => {
                    const radius = baseRadius + index * baseRadius;
                    const opacity = 0.8 - index * 0.1;
                    return (
                        <div
                            key={index}
                            className="absolute rounded-full border overflow-hidden"
                            style={{
                                width: `${radius}px`,
                                height: `${radius}px`,
                                borderWidth: '1px',
                                borderColor: `rgba(228, 231, 236, ${opacity})`,
                                top: `calc(50% - ${radius / 2}px)`,
                                left: `calc(50% - ${radius / 2}px)`
                            }}
                            data-testid={`${testId}-circle-${index}`}
                        />
                    );
                })}
                
                {logoSrc && (
                    <div
                        className="flex items-center justify-center"
                        data-testid={`logo-${testId}-container`}
                    >
                        <img
                            src={logoSrc}
                            alt={logoAlt}
                            data-testid={`logo-${testId}`}
                        />
                    </div>
                )}
            </div>
        </div>
    );
};