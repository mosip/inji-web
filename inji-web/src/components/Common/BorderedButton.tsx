import React from "react";

export const BorderedButton:React.FC<BorderedButtonProps> = (props) => {
  return <div className={"bg-gradient-to-r rounded-lg from-iw-primary to-iw-secondary p-0.5"}>
            <div className="py-2 px-4 rounded-lg bg-white">
                <button
                    data-testid={props.testId}
                    onClick={props.onClick}
                    className="text-iw-tertiary font-bold rounded-lg">
                    {props.title}
                </button>
            </div>
    </div>
}

export type BorderedButtonProps = {
    testId: string;
    onClick: ()=> void;
    title: string;
}
