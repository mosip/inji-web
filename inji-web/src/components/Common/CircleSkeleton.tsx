type CircleSkeletonProps = {
    size?: string;
    className?: string;
};

export const CircleSkeleton = ({
                                   size = 'w-20 h-20',
                                   className = '',
                               }: CircleSkeletonProps) => {
    return (
        <div
            className={`rounded-full bg-gray-300 animate-pulse ${size} ${className}`}
        />
    );
};
  