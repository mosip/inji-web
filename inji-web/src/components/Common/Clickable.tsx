import React from 'react';

export interface ClickableProps {
    onClick: () => void;
    children: React.ReactNode;
    testId: string;
    className: string;
}

export const Clickable: React.FC<ClickableProps> = ({
                                                        onClick,
                                                        children,
                                                        testId,
                                                        className = '',
                                                    }) => {
    const handleKeyPress = (event: React.KeyboardEvent<HTMLDivElement>) => {
        if (event.key === 'Enter' || event.key === ' ') {
            event.preventDefault();
            onClick();
        }
    };

    return (
        <div
            role="menuitem"
            tabIndex={0}
            className={className}
            onClick={onClick}
            onKeyDown={handleKeyPress}
            data-testid={testId}
        >
            {children}
        </div>
    );
};
