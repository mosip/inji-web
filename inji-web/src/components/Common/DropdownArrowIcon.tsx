import React from 'react';
import {RiArrowUpSFill, RiArrowDownSFill} from 'react-icons/ri';
import { DropdownArrowIconProps } from '../../types/data';
import { GradientWrapper } from './GradientWrapper';

const DropdownArrowIcon: React.FC<DropdownArrowIconProps> = ({isOpen, size = 20, color = 'var(--iw-color-languageArrowIcon)'}) => {
    const ArrowIcon = isOpen ? RiArrowUpSFill : RiArrowDownSFill;

    return (
        <GradientWrapper>
            <ArrowIcon size={size} color={color} />
        </GradientWrapper>
    );
};

export default DropdownArrowIcon;
