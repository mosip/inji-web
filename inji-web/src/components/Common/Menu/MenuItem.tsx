import React from "react";
import {MenuStyles} from "./MenuStyles";

interface MenuItemProps {
    label: string;
    onClick: () => void;
    testId: string;
    icon?: React.ReactNode;
    color?: string;
}

export const MenuItem: React.FC<MenuItemProps> = ({label, onClick, testId, icon, color}) => {
    return (
        <button
            onClick={(event: React.MouseEvent) => {
                event.stopPropagation();
                onClick()
            }}
            className={MenuStyles.menuItem.base}
            data-testid={`menu-item-${testId}`}
            role="menuitem"
        >
            <div className={MenuStyles.menuItem.menuItemWrapper}>
                {icon && <span className={MenuStyles.menuItem.icon}>{icon}</span>}
                <span data-testid={`label-${testId}`} className={MenuStyles.menuItem.label}
                      style={{color: color}}>{label}</span>
            </div>
        </button>
    );
};
