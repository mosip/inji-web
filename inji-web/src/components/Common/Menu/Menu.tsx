import React, {useEffect, useRef, useState} from "react";
import {MenuItem} from "./MenuItem";
import {MenuItemType} from "../../../types/data";
import {MenuStyles} from "./MenuStyles";

interface MenuProps {
    triggerComponent: React.ReactNode;
    menuItems: MenuItemType[];
    testId: string;
}

export const Menu: React.FC<MenuProps> = ({triggerComponent, menuItems, testId}) => {
    const [isOpen, setIsOpen] = useState(false);
    const buttonRef = useRef<HTMLButtonElement>(null);
    const menuRef = useRef<HTMLDivElement>(null);

    useEffect(() => {
        const handleClickOutside = (event: MouseEvent) => {
            if (
                menuRef.current &&
                !menuRef.current.contains(event.target as Node) &&
                buttonRef.current &&
                !buttonRef.current.contains(event.target as Node)
            ) {
                setIsOpen(false);
            }
        };

        if (isOpen) {
            document.addEventListener("mousedown", handleClickOutside);
        }

        return () => {
            document.removeEventListener("mousedown", handleClickOutside);
        };
    }, [isOpen]);


    return (
        <div className={MenuStyles.menu.container} data-testid={`${testId}-menu`}>
            <button
                className={MenuStyles.menu.button}
                ref={buttonRef}
                onClick={(event) => {
                    event.stopPropagation();
                    setIsOpen((prev) => !prev);
                }}
                data-testid={`icon-${testId}-menu`}
            >
                {triggerComponent}
            </button>

            {isOpen && (
                <div
                    ref={menuRef}
                    role="menu"
                    data-testid={`menu-${testId}`}
                    className={MenuStyles.menu.menuContainer}
                    style={{zIndex: 9999}}
                >
                    <div className={MenuStyles.menu.caretOuter}></div>
                    <div className={MenuStyles.menu.caretInner}></div>

                    {menuItems.map((item) => (
                        <MenuItem
                            key={item.id}
                            label={item.label}
                            icon={item.icon}
                            color={item.color}
                            onClick={() => {
                                item.onClick();
                                setIsOpen(false);
                            }}
                            testId={item.id}
                        />
                    ))}
                </div>
            )}
        </div>
    );
};
