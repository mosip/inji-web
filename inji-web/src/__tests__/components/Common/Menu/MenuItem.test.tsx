import React from 'react';
import {fireEvent, render, screen} from '@testing-library/react';
import {MenuItem} from '../../../../components/Common/Menu/MenuItem';
import {Clickable} from "../../../../components/Common/Clickable";

describe('MenuItem Component', () => {
  const mockOnClick = jest.fn();
  const label = 'Test Item';
  const testId = 'test-menu';

  beforeEach(() => {
    mockOnClick.mockClear();
  });

  it('renders correctly and matches snapshot', () => {
    const { asFragment } = render(
      <MenuItem label={label} onClick={mockOnClick} testId={testId} />
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders with the correct label', () => {
    render(<MenuItem label={label} onClick={mockOnClick} testId={testId} />);

    const menuItem = screen.getByTestId("menu-item-test-menu");
    expect(menuItem).toHaveTextContent(label);
    expect(screen.getByTestId(`label-${testId}`)).toHaveTextContent(label);
  });

  it('renders with the correct role', () => {
    render(<MenuItem label={label} onClick={mockOnClick} testId={testId} />);

    expect(screen.getByRole('menuitem')).toBeInTheDocument();
  });

  it('calls onClick when clicked', () => {
    render(<MenuItem label={label} onClick={mockOnClick} testId={testId} />);

    const menuItem = screen.getByTestId("menu-item-test-menu");
    fireEvent.click(menuItem);

    expect(mockOnClick).toHaveBeenCalledTimes(1);
  });

  it('stops event propagation when clicked', () => {
    const mockContainerClick = jest.fn();

    render(
      <Clickable onClick={mockContainerClick} testId={`${testId}-wrapper`}>
        <MenuItem label={label} onClick={mockOnClick} testId={testId} />
      </Clickable>
    );

    const menuItem = screen.getByTestId("menu-item-test-menu");
    fireEvent.click(menuItem);

    expect(mockOnClick).toHaveBeenCalledTimes(1);
    expect(mockContainerClick).not.toHaveBeenCalled();
  });

  it('renders without testId when not provided', () => {
    render(<MenuItem label={label} onClick={mockOnClick} testId={"delete"}/>);

    const menuItem = screen.getByRole('menuitem');

    expect(menuItem).toHaveAttribute('data-testid', 'menu-item-delete');
  });
});