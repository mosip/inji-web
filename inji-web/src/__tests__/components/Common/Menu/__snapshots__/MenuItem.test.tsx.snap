// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MenuItem Component renders correctly and matches snapshot 1`] = `
<div>
  <button
    class="w-full text-left px-4 py-2 text-sm text-gray-700 hover:bg-gray-100"
    data-testid="test-menu-item"
    role="menuitem"
  >
    Test Item
  </button>
</div>
`;
